src/jsonstream.cpp

```diff
@@ -82,10 +101,13 @@ int JsonStream::json_parse(ParsedJson &pj) {
             }
             size_t last_index = find_last_json_buf_idx(_buf, _batch_size, pj);
```

Comment:
Is `if (last_index == pj.n_structural_indexes)` acceptable here? It indicates that `last_index` is an out of range response, and then you wouldn't need multiple branches. It's true that in this case 0 is the only out of range response you can possibly get, but it still works the same:

```c++
if (last_index == pj.n_structural_indexes) {
  pj.error_code = simdjson::EMPTY;
  return pj.error_code;
}
pj.n_structural_indexes = last_index + 1;
```

Reply:
Ok. So this is a hack again. If there is just one document, then find_last_json_buf_idx will run through it and set last_index == 0. However, pj.n_structural_indexes might be arbitrarily large (certainly not zero).

I will open an issue.

Reply:
Issue opened https://github.com/lemire/simdjson/issues/470

Reply:
Ah, so we'd need to change find_last_json_buf_idx to return pj.n_structural_indexes at that point. (Which, frankly, we probably have to do more work to return 0 :))

Reply:
@jkeiser Sadly, if you have one full document, you cannot tell that it is the case before you scan it out. But, yeah, it should be doable to return pj.n_structural_indexes or something instead of hacking things out like I am doing here.